lib/ansible/module_utils/facts.py

```diff
@@ -3067,15 +3070,33 @@ def get_file_lines(path):
     return ret
 
 def ansible_facts(module):
+    # Retrieve module parameters
+    gather_subset = [ 'all' ]
+    if 'gather_subset' in module.params:
+        gather_subset = module.params['gather_subset']
+
+    # Retrieve all facts elements
+    if 'all' in gather_subset:
+        gather_subset = [ 'min', 'hardware', 'network', 'virtual' ]
```

Comment:
Might be more likely for code to stay in sync if we derive this from ALLOWED_FACT_SUBSET.  Maybe like this:
``` python
if 'all' in gather_subset:
    gather_subset = ALLOWED_FACT_SUBSET.difference(('all',))